File: Quality Adjusted Life/QualityAdjustedLife.cpp
/*
 Tyler Huffman
 2020-08-08
 https://open.kattis.com/problems/qaly
*/
#include <iostream>
using namespace std;

int main(){
	int counter;       // Stores the number of cases
	float total = 0.0; // Holds the sum of the products
	float temp1;       // Holds the first number
	float temp2;       // Holds the second number
	
	cin >> counter;    // Grab the number of iterations
	// Use the counter to iterate through the list of inputs
	for (counter; counter > 0; counter -= 1){ 
		cin >> temp1;             // Grab the quality
		cin >> temp2;             // Grab the years at said quality
		total += (temp1 * temp2); // Add the product of the years and quality to the total score 
		}
	cout << total << endl; // Display the total quality
	return 0;}

File: Quality Adjusted Life/QualityAdjustedLife.py
# Darren Good, Tyler Huffman
# 2020-03-04
# https://open.kattis.com/problems/qaly

from sys import stdin
# Grab the input (x,y), store the product of the input into a list and find the sum of the list  
print(sum([float(inp_str.split()[0]) * float(inp_str.split()[1]) for inp_str in stdin.read().strip().split('\n')[1:]]))